### PR TITLE
Feat/chart component

### PR DESCRIPTION
**Pull request by dgorkavyi**

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -26,6 +28,7 @@
   },
   "devDependencies": {
     "@types/node": "^12.20.16",
+    "@types/query-string": "^6.3.0",
```
Comment (yevhenii-moroziuk) [inline frontend/package.json]: no need for this definition since it doesn't work

File: frontend/src/components/Chart/Chart.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import ChartComponent from 'react-chartjs-2';
+import React from 'react';
+
+const testData = {
```
Comment (yevhenii-moroziuk) [inline frontend/src/components/Chart/Chart.tsx]: move the test data out the component

File: frontend/src/components/Chart/Chart.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      label: '# of Votes',
+      data: [12, 19, 3, 5, 2, 3],
+      fill: false,
+      backgroundColor: 'rgb(255, 99, 132)',
+      borderColor: 'rgba(255, 99, 132, 1)'
+    }
+  ]
+};
+
+type TDataSet = [
```
Comment (yevhenii-moroziuk) [inline frontend/src/components/Chart/Chart.tsx]:
```
type DataItem = {
    label: string;
    data: number[];
    fill?: boolean;
    backgroundColor?: string;
    borderColor?: string;
  }
```

Do not use I prefix for Types and interfaces

File: frontend/src/components/Chart/Chart.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  | 'radar'
+  | 'doughnut'
+  | 'polarArea'
+  | 'bubble'
+  | 'pie'
+  | 'scatter';
+
+type IChartProps = {
+  type?: TChartType;
+  data?: TDataSet | any;
```
Comment (yevhenii-moroziuk) [inline frontend/src/components/Chart/Chart.tsx]: Please use the interface here for code consistency. Do not use any and make type and data required.

File: frontend/src/components/Chart/Chart.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+const ChartExample: React.FC<IChartProps> = ({ type, data }) => {
+  console.log('hello');
+  return (
+    <>
+      <ChartComponent type={type || 'line'} data={data || testData} />
+    </>
+  );
+};
+
```
Comment (yevhenii-moroziuk) [inline frontend/src/components/Chart/Chart.tsx]: Rename to Chart, remove console.log and react fragment.

**Review by yevhenii-moroziuk — CHANGES_REQUESTED** [review]

File: frontend/src/components/Chart/Chart.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  | 'pie'
+  | 'scatter';
+
+interface ChartProps {
+  type?: ChartType;
+  labels?: string[];
+  dataSets?: DataItem[];
+}
+
+const Chart: React.FC<ChartProps> = ({ type, dataSets, labels }) => (
```
Comment (turfle) [inline frontend/src/components/Chart/Chart.tsx]: nit: it would be good to set default values to optional props or make them required

Comment (dgorkavyi) [reply frontend/src/components/Chart/Chart.tsx]: Sorry, forgot to remove the "?" in the interface ChartProps. All of them are needed. By the way if you need some default values to see how it works, you can get it from testprops.ts

Comment (yevhenii-moroziuk) [reply frontend/src/components/Chart/Chart.tsx]: Let's add optional property options and use defaultProps as default. Please try to create an interface based on documentation for options

**Review by yevhenii-moroziuk — APPROVED** [review]

**Comment by dgorkavyi** [comment]
Conflicts were fixed, the merge was done from the terminal, the commit is in the logs

**Pull request closed without merging**